Emby.Server.Implementations/AppBase/ConfigurationHelper.cs

```diff
@@ -48,8 +49,9 @@ public static object GetXmlConfiguration(Type type, string path, IXmlSerializer
             // If the file didn't exist before, or if something has changed, re-save
             if (buffer == null || !newBytes.AsSpan(0, newBytesLen).SequenceEqual(buffer))
             {
-                Directory.CreateDirectory(Path.GetDirectoryName(path));
+                var directory = Path.GetDirectoryName(path) ?? throw new ResourceNotFoundException(nameof(path));
```

Comment:
Looking at this closer, I think an `IllegalArgumentException` would be better, because the  exception is thrown when the provided path isn't a valid path for a directory, not when the directory doesn't exist.